src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -249,6 +282,24 @@ void concat_in_place_optimization::optimize_cascade(concatenation_node& node, st
     update_in_place_concat_paddings(concat_layout, preds_layouts, node.get_primitive()->axis, false);
     size_t i = 0;
     for (auto& dep : node.get_dependencies()) {
+        if (dep.first->is_type<reshape>()) {
+            auto reshape_node = dep.first;
+            bool is_dynamic = reshape_node->is_dynamic();
+            bool is_planar = format::is_default_format(reshape_node->get_output_layout().format);
+            bool no_pad = !reshape_node->get_output_layout().data_padding && !reshape_node->get_input_layouts().empty()
+                                && !reshape_node->get_input_layout(0).data_padding;
+            // if reshape is the input of concat and it is optimized out,
+            // update padding of dependency node of reshape
+            if (is_dynamic && is_planar && no_pad && !reshape_node->is_output() && !reshape_node->has_fused_primitives()) {
+                // reshape node is set to optimized out.
+                reshape_node->can_be_optimized(true);
```

Comment:
1. What is the policy? 
a. support [optimized reshape] + concat only. 
- a.1 Here target reshape is opt out in the previous pass. 
- a.2. Even though reshape is not marked as optimized out from the previous pass, mark it as optimize here. 
b. support both [optimized reshape] + concat and [not optimized reshape] + concat
- b.1 same as a1
- b2 same as a2

Reply:
Fixed it by offline comments. thanks.